File: user-service/src/main/java/com/ilogos/user/response/SuccessResponse.java
package com.ilogos.user.response;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class SuccessResponse<T> extends AbstractResponse {
    @Schema(description = "Response data")
    private T data = null;

    public static ResponseEntity<SuccessResponse<?>> response() {
        return ResponseEntity.status(HttpStatus.OK).body(new SuccessResponse<>());
    }

    public static <T> ResponseEntity<SuccessResponse<T>> response(T data) {
        return ResponseEntity.status(HttpStatus.OK).body(new SuccessResponse<>(data));
    }

    public static <T> ResponseEntity<SuccessResponse<T>> response(HttpStatus status, T data) {
        return ResponseEntity.status(status).body(new SuccessResponse<>(data));
    }
}

File: course-service/src/main/kotlin/com/ilogos/course/response/SuccessResponse.kt
package com.ilogos.course.response

import io.swagger.v3.oas.annotations.media.Schema
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity

class SuccessResponse<T>(@Schema(description = "Response data") val data: T?) : AbstractResponse() {

    companion object {
        fun response(): ResponseEntity<SuccessResponse<Any>> {
            return ResponseEntity.status(HttpStatus.OK).body(SuccessResponse<Any>(null))
        }

        fun <T> response(data: T): ResponseEntity<SuccessResponse<T>> {
            return ResponseEntity.status(HttpStatus.OK).body(SuccessResponse(data))
        }

        fun <T> response(status: HttpStatus, data: T): ResponseEntity<SuccessResponse<T>> {
            return ResponseEntity.status(status).body(SuccessResponse(data))
        }
    }
}
